browser/ui/webui/new_tab_page/brave_new_tab_ui.cc

```diff
@@ -35,6 +38,15 @@ BraveNewTabUI::BraveNewTabUI(content::WebUI* web_ui, const std::string& name)
           web_ui,
           true /* Needed for legacy non-mojom message handler */),
       page_factory_receiver_(this) {
+  content::WebContents* web_contents = web_ui->GetWebContents();
+  CHECK(web_contents);
+
+  const bool was_restored =
+      web_contents->GetController().GetLastCommittedEntry()->IsRestored();
```

Comment:
I'm not sure about GetLastCommittedEntry() value in the BraveNewTabUI constructor. Maybe it is safer to check its value by
```
if (const auto* last_committed_entry = web_contents->GetController().GetLastCommittedEntry()) {
  was_restored = last_committed_entry->IsRestored();
}

```